redwood-cli/src/main/kotlin/app/cash/redwood/generator/diffConsumingGeneration.kt

```diff
@@ -37,52 +36,40 @@ import com.squareup.kotlinpoet.TypeSpec
 import com.squareup.kotlinpoet.asTypeName
 
 /*
-public class DiffConsumingSunspotWidgetFactory<W : Any>(
-  private val delegate: SunspotWidgetFactory<W>,
+public class SunspotDiffConsumingNodeFactory<W : Any>(
+  private val widgets: SunspotWidgetFactory<W>,
   private val json: Json = Json.Default,
   private val mismatchHandler: ProtocolMismatchHandler = ProtocolMismatchHandler.Throwing,
 ) : DiffConsumingNode.Factory<W> {
-  override val RedwoodLayout = DiffConsumingRedwoodLayoutWidgetFactory(delegate.RedwoodLayout, json, mismatchHandler)
-
   override fun create(
     parentId: Id,
     parentChildren: Widget.Children<W>,
     tag: WidgetTag,
-  ): DiffConsumingWidget<W>? = when (tag.value) {
-    1 -> SunspotBox(parentId, parentChildren)
-    2 -> SunspotText(parentId, parentChildren)
-    3 -> SunspotButton(parentId, parentChildren)
-    1_000_001 -> RedwoodLayout.Row(parentId, parentChildren)
-    1_000_002 -> RedwoodLayout.Column(parentId, parentChildren)
+  ): DiffConsumingNode<W>? = when (tag.value) {
+    1 -> DiffConsumingTextNode(parentId, parentChildren, delegate.Text(), json, mismatchHandler)
+    2 -> DiffConsumingButtonNode(parentId, parentChildren, delegate.Button(), json, mismatchHandler)
+    1_000_001 -> DiffConsumingRedwoodLayoutRowNode(parentId, parentChildren, delegate.RedwoodLayout.Row(), json, mismatchHandler)
+    1_000_002 -> DiffConsumingRedwoodLayoutColumnNode(parentId, parentChildren, delegate.RedwoodLayout.Column(), json, mismatchHandler)
```

Comment:
This function may eventually blow bytecode limits but I'm inclined to not do anything for now. When it does we can split it into multiple private functions and first match on the schema tag (millions decimal place) and then the widget tag.